cabal-install/src/Distribution/Client/Init/Heuristics.hs

```diff
@@ -156,9 +160,23 @@ scanForModulesIn projectRoot srcRoot = scan srcRoot []
     ignoreDir ('.':_)  = True
     ignoreDir dir      = dir `elem` ["dist", "_darcs"]
 
+-- | Read the contents of the handle and parse for Language pragmas and
+-- and other module names that might be part of this project.
 findImportsAndExts :: FilePath -> SourceFileEntry -> IO SourceFileEntry
-findImportsAndExts projectRoot sf = do
-  s <- readFile (sfToFileName projectRoot sf)
+findImportsAndExts projectRoot sf =
+    -- Avoid resource exhaustion, see https://github.com/haskell/cabal/issues/5115
+    bracket
```

Comment:
this is just `withFile`, isn't it?

Reply:
FWIW, I'd simply read the file with `Data.ByteString.readFile`.

Reply:
Thank you for the review!

Reply:
And then decode the file to utf8 manually, I presume? Or re-write the parsing to use `ByteString` instead and decode only the final result?

Reply:
`fromUTF8BS` is simplest way forward. i would bother rewriting the parser bits. Here performance is not critical (we only need to be careful with scarse resources = file descriptors)